packages/hebao_v1/contracts/stores/SecurityStore.sol

```diff
@@ -65,15 +82,20 @@ contract SecurityStore is DataStore
         return wallets[wallet].guardians.length;
     }
 
-    function addGuardian(address wallet, address guardian)
+    function addGuardian(address wallet, address guardian, uint info)
         public
         onlyManager
     {
         require(guardian != address(0), "ZERO_ADDRESS");
         Wallet storage w = wallets[wallet];
         require(w.guardianIdx[guardian] == 0, "GUARDIAN_EXISTS");
 
-        w.guardians.push(guardian);
+        Guardian memory g = Guardian(
```

Comment:
if we allow a guardian to have different roles, then maybe we should do:

`info |= existing.info` and remove the `require` above.

Reply:
When updating a guardian the value is currently just overwritten (as `|` can only add bits, which would be a bit strange).

Reply:
somewhere we may want to make it clear the types are overridden. And maybe we need to change `types` to `type` as there won't be more than one type.